plugins/inputs/modbus/modbus.go

```diff
@@ -88,19 +95,23 @@ const sampleConfig = `
 
   # TCP - connect via Modbus/TCP
   controller = "tcp://localhost:502"
- 
+
   ## Serial (RS485; RS232)
   # controller = "file:///dev/ttyUSB0"
   # baud_rate = 9600
   # data_bits = 8
   # parity = "N"
   # stop_bits = 1
+  # transmission_mode = "RTU"
+
+  ## Trace the connection to the modbus device as debug messages
```

Comment:
How is this different than running all of telegraf with debug logging? Is it the same output, but just for this plugin?

Reply:
My understanding is that the usual `debug` mode enables all debug output concerning the **telegraf** plugin. However, with this option you can enable the (connection) debug output of the **underlying library**. This underlying library might (and actually does) output a huge amount of messages which buries the "usual" debugging output of telegraf under tons of raw modbus connection messages rendering it useless. Nevertheless, it is sometimes useful to also see those underlying raw data to debug (connection) problems.
Therefore the idea of this option is to enable another debug level beyond the usual telegraf debugging.